### PR TITLE
Issue #1454 #1476 added check for default network and correct flag for kvm driver

### PR DESCRIPTION
**Pull request by anjannath**

### CODE REVIEW

File: cmd/minishift/cmd/start_preflight.go [hunk 1]
```diff
@@ -58,6 +58,12 @@ func preflightChecksBeforeStartingHost() {
 			"Checking if KVM driver is installed",
 			false, configCmd.WarnCheckXHyveDriver.Name,
 			driverErrorMessage)
+		preflightCheckSucceedsOrFails(
+			configCmd.SkipCheckKVMDriver.Name,
+			checkDefaultNetwork,
+			"Checking if default network is present",
+			false, configCmd.WarnCheckXHyveDriver.Name,
```
Comment (gbraad) [inline cmd/minishift/cmd/start_preflight.go]: Xhyve ?

File: cmd/minishift/cmd/start_preflight.go [hunk 2]
```diff
@@ -243,6 +249,20 @@ func checkKvmDriver() bool {
 	return true
 }
 
+//checkDefaultNetwork returns true if the "default" network is present
+func checkDefaultNetwork() bool {
```
Comment (gbraad) [inline cmd/minishift/cmd/start_preflight.go]: `CheckKVMDefaultNetwork` or `CheckLibvirtDefaultNetwork`

File: cmd/minishift/cmd/start_preflight.go [hunk 3]
```diff
@@ -243,6 +249,20 @@ func checkKvmDriver() bool {
...
+	cmd := exec.Command("virsh", "--connect", "qemu:///system", "net-list")
+	stdOutStdError, err := cmd.CombinedOutput()
+	if err != nil {
+		return false
+	}
+	stdOut := fmt.Sprintf("%s", stdOutStdError)
+	if strings.Contains(stdOut, "default") {
```
Comment (gbraad) [inline cmd/minishift/cmd/start_preflight.go]: can you make sure that it is only `default`, and not `mydefault` ? and also , that the network is active?

```
$ virsh --connect qemu:///system net-list
 Name                 State      Autostart     Persistent
----------------------------------------------------------
 default              active     yes           yes
 docker-machines      active     yes           yes
 vagrant-libvirt      active     no            yes
```

File: cmd/minishift/cmd/start_preflight.go (hunk 1 above)
Comment (gbraad) [reply cmd/minishift/cmd/start_preflight.go]: which default network.... can this be more specific?
I see people think... `default` network? I know of default gateway?!

**Comment by gbraad** [comment]
If possible, can you also add a commit for #1472 to test for `virsh`?

File: cmd/minishift/cmd/start_preflight.go (hunk 1 above)
Comment (anjannath) [reply cmd/minishift/cmd/start_preflight.go]: i saw the other check also used  `configCmd.WarnCheckXHyveDriver.Name` so..

File: cmd/minishift/cmd/start_preflight.go (hunk 3 above)
Comment (anjannath) [reply cmd/minishift/cmd/start_preflight.go]: i added the check for `active` but WDYT? should it be in a different function?

Comment (gbraad) [reply cmd/minishift/cmd/start_preflight.go]: haven't checked yet, but from my feeling breaking this up feels unnecessary

File: cmd/minishift/cmd/start_preflight.go [hunk 4]
```diff
@@ -58,6 +59,12 @@ func preflightChecksBeforeStartingHost() {
 			"Checking if KVM driver is installed",
 			false, configCmd.WarnCheckXHyveDriver.Name,
 			driverErrorMessage)
+		preflightCheckSucceedsOrFails(
+			configCmd.SkipCheckKVMDriver.Name,
+			checkLibvirtDefaultNetwork,
+			"Checking if Libvirt default network is present and active",
+			false, configCmd.WarnCheckXHyveDriver.Name,
```
Comment (gbraad) [inline cmd/minishift/cmd/start_preflight.go]: Still xhyve? :-O

**Comment by gbraad** [comment]
add to whitelist

**Comment by gbraad** [comment]
retest this please

**Pull request merged**